src/ssl.c

```diff
@@ -13303,6 +13306,7 @@ int wolfSSL_DTLS_SetCookieSecret(WOLFSSL* ssl,
 
         case FIRST_REPLY_THIRD :
             if ( (ssl->error = SendChangeCipher(ssl)) != 0) {
+                ProcessReplyEx(ssl, 1); /* See if an alert was sent. */
```

Comment:
I think there should a guard around these new calls. This can be the new default but after hearing from @toddouska it might be good to allow users to revert back to the previous functionality. Also let's make sure this gets into the release notes.

Reply:
I don't recall Todd saying something that implied a guard was needed. What is your concern? Agree that this should be in the release notes.

Reply:
I agree with @julek-wolfssl , there should be a macro guard option. Enable by default, but option to disable. My concern is around it trying to do a read on a closed socket and behavior variations between network stacks.

Reply:
He commented that some users expect data to travel in one direction at a time although this pr focusses on the handshake process so data is travelling in both directions. It might not apply in this case.

My main concern is is it safe to read from a closed/failed socket? It probably is since it will just return the error again (though worst case it could hang). I'll run some tests out of curiosity.

Reply:
Fair point. I'll add in a macro guard that's enabled by default.

Reply:
And remember to document this guard at the top of `ssl.c`.

Reply:
Many embedded users will have custom read and write callbacks and they might not be ready for this change.

Reply:
> Many embedded users will have custom read and write callbacks and they might not be ready for this change.

In that case, I'll make the macro guard on by default.

Reply:
> And remember to document this guard at the top of `ssl.c`.

I found a section at the top of tls13.c for that, but not in ssl.c. Am I just missing it or does it only exist in tls13.c?

Reply:
Looks like currently only `OPENSSL_COMPATIBLE_DEFAULTS` is documented in `ssl.c`. I thought I remembered more macros being documented there. Might as well start now.

Reply:
> When a stream socket peer has performed an orderly shutdown, the 
return value will  be  0  (the  traditional "end-of-file" return).

So this is what I'm getting. This should work on most major OS's. I'd be 
worried that embedded `RECV_FUNCTION` implementations might go crazy 
when called in an error state.

> In that case, I'll make this off by default (with the macro guard).

I think that this might be the safest way forward.